### ISSUE TITLE
File descriptor leak

### ISSUE THREAD

**Issue by dion-conqa** [cloudant/couchbackup#535]
Please [read these guidelines](http://ibm.biz/cdt-issue-guide) before opening an issue.

## Bug Description
There is a file descriptor leak in this library. We found that after calling `backup` there would be a file left open in the `/tmp` folder.

We run our backup process from an AWS lambda that has a hard limit of 1,024 file descriptors.  Since we are using a lambda, the process doesn't exit straight away and gets reused for the next call.  

Each of our projects is a separate database meaning that we slowly have more databases to back up and In August, we started getting error messages like
`EMFILE: too many open files, open '/tmp/couchbackup/couchbackup-qa_xsxxxxxxxxxxxx-xxxxxxxxx_2022-11-15T17:47:52.144Z.log'` 

This issue is cause by the [apiDefaults](https://github.com/cloudant/couchbackup/blob/f44a8a3fe2967cd33e32dd911b699f019bbee1a4/includes/config.js#L27) function.  When you look into the [`tmp.fileSync()`](https://github.com/raszi/node-tmp/blob/da3bc1ac9a7880def4a3129c4f8bfe96a63bf99c/lib/tmp.js#L171) function you can see that it opens a file which is only closed if you set `opts.discardDescriptor` to true. 

Given that the code is only after a tmp file name it might be better to use [`tmpNameSync`](https://github.com/raszi/node-tmp/blob/da3bc1ac9a7880def4a3129c4f8bfe96a63bf99c/lib/tmp.js#L100)

You can see the number of open fd using Lambda Insights.  This is the graph with the current library.
![image](https://user-images.githubusercontent.com/103007470/206011611-8c2c8538-ae69-4ab4-9595-4c6a2929f56c.png)

After I changed that call to be `tmp.fileSync({ discardDescriptor: true }).name` 
![image](https://user-images.githubusercontent.com/103007470/206011914-9eb0e40f-4579-4954-a7eb-3ba354464d8d.png)

### 1. Steps to reproduce and the simplest code sample possible to demonstrate the issue

This isn't a lambda but you can see the files being left open.  They will be finally closed when the process exists, but AWS Lambda leaves the process running so it can be reused. 
```js
import { exec } from "child_process";
import { join } from "path";
import os from "os";
import fs from "fs";
import { backup } from "@cloudant/couchbackup";

const listFileDescriptor = () => {
  return new Promise((resolve, reject) => {
    exec(`ls -l /proc/${process.pid}/fd`, (err, stdout, stderr) => {
      if (err) {
        reject(err);
      } else {
        const parts = stdout.split("\n");
        // there will be other fd open, but it's only the tmp- ones causing problems
        const tmpFiles = parts.filter((x) => x.includes("/tmp/"));
        resolve(tmpFiles);
      }
    });
  });
};

const databases = [
    "https://user:password@host/database1",
    "https://user:password@host/database2",
  ];

const backupDatabases = async () => {
  const fdStart = await listFileDescriptor();
  console.log(fdStart);

  const tmpFolder = os.tmpdir();
  const tmpFile = join(
    tmpFolder,
    `couchbackup-${new Date().toISOString()}.log`
  );

  await backup(databases[0], fs.createWriteStream("dump1.json"), (err) => {
    if (err) {
      return new Error(err, "CouchBackup process failed");
    }
    console.log("couchbackup to file done;");
  });

  await backup(databases[1], fs.createWriteStream("dump1.json"), (err) => {
    if (err) {
      return new Error(err, "CouchBackup process failed");
    }
    console.log("couchbackup to file done;");
  });

  const fdEnd = await listFileDescriptor();
  console.log(fdEnd);
};

backupDatabases();


```

### 2. What you expected to happen
There should be no temp files when the `backup` call has completed and the number of open fd should not be increasing. 

### 3. What actually happened
There are two temp files left open.
![image](https://user-images.githubusercontent.com/103007470/206013546-82f13b12-831f-4301-b2c8-beba7ff3f69e.png)

## Environment details

- Version(s) that are affected by this issue.
    > 2.9.3
- Node.js version
    >14.20.0

**Comment by emlaver** [comment]
@dion-conqa thanks for sharing these details.  I've managed to replicate this issue locally using your your code against the latest `couchbackup` version.  We'll investigate further and provide any updates on this ticket.

**Comment by emlaver** [comment]
We've merged a fix in #536 and it will be in the next release.

**Comment by dion-conqa** [comment]
Thank you.
